boot/efi_amd64.go

```diff
@@ -0,0 +1,52 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+// +build amd64
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package boot
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+var (
+	efiVarDir = "/sys/firmware/efi/vars"
```

Comment:
Looking at the kernel bits CONFIG_EFIVAR_FS gives `/sys/firmware/efi/efivars` and AFAICT is the currently recommended way of accessing those variables. There's an older CONFIG_EFI_VARS that would expose `/sys/firmware/efi/vars`. Looks like CONFIG_EFI_VARS is disabled in Fedora, RHEL, Arch, but enabled in Ubuntu (actually both are enabled?). Some additional notes on efivars here: https://www.kernel.org/doc/Documentation/filesystems/efivarfs.txt

It looks like we should try both, with efivars being first, and fallback to `efi/vars` iff the former does not exist?

Reply:
Also note that the format of `efi/vars` and `efi/efivars` is different, entries are directories in `efi/vars` and files in `efi/efivars`. If only one is to be implemented `efi/efivars` seems to be more future-proof.

Reply:
yes we can try both, thanks for checking on the kernel docs about this